NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
-                                iS, propagation.solution_types[self._raytracer.get_solution_type(iS)], viewing_angle / units.deg, (viewing_angle - cherenkov_angle) / units.deg))
-                            delta_Cs.append(delta_C)
-
-                        # discard event if delta_C (angle off cherenkov cone) is too large
-                        if min(np.abs(delta_Cs)) > self._cfg['speedup']['delta_C_cut']:
-                            logger.debug('delta_C too large, event unlikely to be observed, skipping event')
+                            sim_station = calculate_sim_efield_for_emitter(emitters=event_group.get_sim_emitters(),
+                                                sid=sid, cid=channel_id,
+                                                det=self._det, propagator=self._propagator, medium=self._ice, config=self._config,
+                                                rnd=self._rnd, antenna_pattern_provider=self._antenna_pattern_provider,
+                                                min_efield_amplitude=float(self._config['speedup']['min_efield_amplitude']) * self._Vrms_efield_per_channel[sid][channel_id],
+                                                time_logger=self.__time_logger)
+                        # skip to next channel if the efield is below the speed cut
+                        if len(sim_station.get_electric_fields()) == 0:
+                            logger.info(f"Eventgroup {event_group.get_run_number()} Station {sid} channel {channel_id:02d} has {len(sim_station.get_electric_fields())} efields, skipping to next channel")
                             continue
 
-                        n = self._raytracer.get_number_of_solutions()
-                        for iS in range(n):  # loop through all ray tracing solution
-                            # skip individual channels where the viewing angle difference is too large
-                            # discard event if delta_C (angle off cherenkov cone) is too large
-                            if np.abs(delta_Cs[iS]) > self._cfg['speedup']['delta_C_cut']:
-                                logger.debug('delta_C too large, ray tracing solution unlikely to be observed, skipping event')
+                        # applies the detector response to the electric fields (the antennas are defined
+                        # in the json detector description file)
+                        apply_det_response_sim(sim_station, self._det, self._config, self.detector_simulation_filter_amp,
+                                            event_time=self._evt_time, time_logger=self.__time_logger,
+                                            detector_simulation_part1=self.detector_simulation_part1)
+                        logger.debug(f"adding sim_station to station {sid} for event group {event_group.get_run_number()}, channel {channel_id}")
+                        station.add_sim_station(sim_station)  # this will add the channels and efields to the existing sim_station object
```

Comment:
One could make this block optional (for the non-trigger-channels) (opt-out) to further reduce the computing time. Although I do not know if that is significant?

Reply:
Also this function is called in the channel loop. Is that correct - efficient? One could first simulate the efields for all channels and than apply to detector response

Reply:
the order shouldn't matter. And we often want the SimChannels for all channels. I think it is even used later to obtain the real channels.